### PR TITLE
refactor: Refactor `aws` provider tests for prefixes

### PR DESCRIPTION
**Pull request by AlinoeDoctari**
Continue the work on #8446

### CODE REVIEW

File: test/unit/lib/plugins/aws/provider.test.js [hunk 1]
```diff
@@ -274,80 +274,6 @@ describe('AwsProvider', () => {
...
-
-    it('should support no prefix', () => {
-      serverless.service.provider.alb = {};
-      serverless.service.provider.alb.targetGroupPrefix = '';
-
-      expect(awsProvider.getAlbTargetGroupPrefix()).to.equal('');
-    });
-  });
-
-  describe('#getAccountInfo()', () => {
```
Comment (pgrzesik) [inline test/unit/lib/plugins/aws/provider.test.js:330]: Is there a replacement test for this removed one? Or it was removed by mistake?

File: test/unit/lib/plugins/aws/provider.test.js [hunk 2]
```diff
@@ -679,6 +605,92 @@ aws_secret_access_key = CUSTOMSECRET
...
+                    ...validBaseEventConfig,
+                    priority: 1,
+                  },
+                },
+              ],
+            },
+          },
+        },
+      });
+      expect(cfTemplate).to.haveOwnProperty('Resources');
```
Comment (pgrzesik) [inline test/unit/lib/plugins/aws/provider.test.js:680]: I believe we can omit that check as if it's missing the test will fail anyway

File: test/unit/lib/plugins/aws/provider.test.js [hunk 3]
```diff
@@ -679,6 +605,92 @@ aws_secret_access_key = CUSTOMSECRET
...
+      const resources = Object.values(cfTemplate.Resources);
```
Comment (pgrzesik) [inline test/unit/lib/plugins/aws/provider.test.js:681]: I think we can skip assignment to this temp variable which is used in only one place - let's go with `Object.values(cfTemplate.Resources).filter` in next line

File: test/unit/lib/plugins/aws/provider.test.js [hunk 4]
```diff
@@ -679,6 +605,92 @@ aws_secret_access_key = CUSTOMSECRET
     });
   });
 
+  describe('#getDeploymentPrefix()', () => {
+    it('should put all artifacts in namespaced folder', async () => {
+      const { cfTemplate } = await runServerless({
+        fixture: 'function',
+        command: 'package',
+      });
+      expect(cfTemplate).to.haveOwnProperty('Resources');
```
Comment (pgrzesik) [inline test/unit/lib/plugins/aws/provider.test.js:614]: same as in comment for `albPrefix` test

File: test/unit/lib/plugins/aws/provider.test.js [hunk 5]
```diff
@@ -679,6 +605,92 @@ aws_secret_access_key = CUSTOMSECRET
...
+      const { cfTemplate } = await runServerless({
+        fixture: 'function',
+        command: 'package',
+        configExt: {
+          provider: {
+            deploymentPrefix: 'test-prefix',
+          },
+        },
+      });
+      expect(cfTemplate).to.haveOwnProperty('Resources');
```
Comment (pgrzesik) [inline test/unit/lib/plugins/aws/provider.test.js:635]: same as in comment for `albPrefix` test

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thank you @AlinoeDoctari - it looks good overall :+1: I just have a few small comments, please let me know what do you think :bow:

File: test/unit/lib/plugins/aws/provider.test.js (hunk 2 above)
Comment (AlinoeDoctari) [reply test/unit/lib/plugins/aws/provider.test.js:680]: Yes, this was a temporary iteration to help nail down the issue, I will remove it in all the tests :+1:

File: test/unit/lib/plugins/aws/provider.test.js (hunk 1 above)
Comment (AlinoeDoctari) [reply test/unit/lib/plugins/aws/provider.test.js:330]: This is a mistake :disappointed: Thanks for catching it :bow:

**Review by pgrzesik — APPROVED** [review]
Looks awesome :raised_hands: Thank you @AlinoeDoctari

**Pull request merged**